kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/S3SinkConnectorIT.java

```diff
@@ -0,0 +1,288 @@
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    waitForConnectorToCompleteSendingRecords(totalNoOfRecordsProduced, FLUSH_SIZE, S3_TEST_BUCKET_NAME);
+
+    // assert records
+    assertEquals(totalNoOfRecordsProduced/FLUSH_SIZE, getNoOfObjectsInS3(S3_TEST_BUCKET_NAME));
+  }
+
+  /**
+   * Case in which bucket permissions/policies are changed while uploading records
+   * Prerequisite : Access key and Secret access key should be set as environment variables
+   * Blocker : Assert Condition
+   * @throws Exception
+   */
+  @Test
+  public void testWithRevokedWritePermissions() throws Exception {
+
+    addReadWritePolicyToBucket(S3_TEST_BUCKET_NAME);
+    KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+
+    // send records to kafka
+    sendRecordsToKafka();
+
+    Map<String, String> props = getConnectorProps();
+    props.put("s3.bucket.name", S3_TEST_BUCKET_NAME);
+    props.put("aws.access.key.id", System.getenv("SECONDARY_USER_ACCESS_KEY_ID"));
+    props.put("aws.secret.access.key", System.getenv("SECONDARY_USER_SECRET_ACCESS_KEY"));
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    Thread.sleep(5000);
+
+    // revoke read/write permission
+    s3RootClient.deleteBucketPolicy(S3_TEST_BUCKET_NAME);
+
+    // produce more records to kafka
+    sendRecordsToKafka();
+    Thread.sleep(10000);
+  }
+
+  @Test
+  public void testWithNetworkInterruption() throws Throwable {
+    //Setup Squid Proxy Container
+    setupSquidProxy();
+
+    // create topics in Kafka
+    KAFKA_TOPICS.forEach(topic -> connect.kafka().createTopic(topic, 1));
+    // send records to kafka
+    sendRecordsToKafka();
+
+    Map<String, String> props = getConnectorProps();
+    props.put("s3.bucket.name", S3_TEST_BUCKET_NAME);
+    props.put(S3SinkConnectorConfig.S3_PROXY_URL_CONFIG,"https://"+squid.getContainerIpAddress() + ":" + squid.getMappedPort(3129));
+
+    // start a sink connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+    // wait for tasks to spin up
+    int minimumNumTasks = Math.min(KAFKA_TOPICS.size(), TASKS_MAX);
+
+    waitForConnectorToStart(CONNECTOR_NAME, minimumNumTasks);
+    waitForConnectorToCompleteSendingRecords(totalNoOfRecordsProduced, FLUSH_SIZE, S3_TEST_BUCKET_NAME);
+
+    // assert records
+    assertEquals(totalNoOfRecordsProduced/FLUSH_SIZE, getNoOfObjectsInS3(S3_TEST_BUCKET_NAME));
+
+    // Shutting down proxy to emulate network unavailability
+    shutdownSquidProxy();
+
+    sendRecordsToKafka();
+    Thread.sleep(30000);
```

Comment:
Should replace with waitForCondition. We should't get the required number of files in S3 bucket.